edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Core/device/NestingInfo.cs

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Hub.Core.Device
+{
+    using System;
+    using System.Threading;
+    using Microsoft.Azure.Devices.Edge.Hub.Core.Identity;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+
+    public class NotNested : INestingInfo
+    {
+        public static INestingInfo Instance { get; } = new NotNested();
+
+        public Option<IIdentity> KnownParent => Option.None<IIdentity>();
+        public bool IsDirectClient => true;
+
+        public void BindToParent(IIdentity parent) => throw new InvalidOperationException("Nesting info is not supported");
+    }
+
+    public class NestingInfo : INestingInfo
```

Comment:
This can be moved to Mqtt project maybe?

Reply:
IDeviceProxy uses it (so amqp also needs to deal with it)